src/lib.rs

```diff
@@ -1359,6 +1391,25 @@ fn perf_map_poller(
         tokens.insert(token, p);
     }
 
+    {
+        // now that the perfmap fd's are registered, we can signal to the main thread that
+        // event polling is ready.
+        let (lock, cvar) = &*polling_signal;
+        let mut polling_is_ready = match lock.lock() {
+            Ok(s) => s,
+            Err(e) => {
+                crit!(
+                    LOGGER.0,
+                    "perf_map_poller(); error grabbing cond mutex: {:?}",
+                    e
+                );
+                return;
```

Comment:
If it can't grab the lock, do we want to fail out of the thread or just continue on? The main thread will time out, so it could still work even with this error.

I'm just wondering what's best - fail if we can't notify the main thread, or continue sending events best-effort?

Reply:
On the success case, I'll notify the thread. On the failure case, I'll log an error and continue.